packages/core/src/types.ts

```diff
@@ -1982,6 +2000,17 @@ export type StateValueFrom<TMachine extends AnyStateMachine> = Parameters<
 export type StateFromMachine<TMachine extends AnyStateMachine> =
   TMachine['initialState'];
 
+export interface ActorSystemInfo {
+  actors: Record<string, AnyActorRef>;
+}
+
+export interface ActorSystem<T extends ActorSystemInfo> {
+  _bookId: () => string;
+  _register: (sessionId: string, actorRef: AnyActorRef) => string;
+  _unregister: (actorRef: AnyActorRef) => void;
+  _set: <K extends keyof T['actors']>(key: K, actorRef: T['actors'][K]) => void;
+  get: <K extends keyof T['actors']>(key: K) => T['actors'][K] | undefined;
+}
```

Comment:
Don't worry about this right now but I find defining a type like this here extremely annoying to work with. It's not exactly an "abstract" type that comes with no direct implementation. It has a single implementation in the `system.ts` and whenever I cmd+click on `ActorSystem` I'm being brought to here by the IDE which isn't that helpful (going to the implementation in the `system.ts` would be way more useful in this instance)

Reply:
Want a class instead?

Reply:
the current runtime code is fine - but the type definition could be derived from the implementation or just declared next to the implementation then jumping to it would at least bring us to the correct file

Reply:
> the current runtime code is fine - but the type definition could be derived from the implementation or just declared next to the implementation then jumping to it would at least bring us to the correct file

I'm open to having this be refactored, if you want to (not immediately sure how to do this)